python/lib/dependabot/python/authed_url_builder.rb

```diff
@@ -4,7 +4,7 @@ module Dependabot
   module Python
     class AuthedUrlBuilder
       def self.authed_url(credential:)
-        token = credential.fetch("token", nil)
+        token = credential.fetch("token", nil)&.gsub("@", "%40")
```

Comment:
Is there something special about the `@` or should we just be using `URI.escape`?

Reply:
`URI.escape` and the likes will also encode the `:` which would make the Basic Auth scheme invalid, so I opted for this instead. I couldn't think of a library method that does exactly what we want.

Reply:
Ah yeah could we split on `:` and escape the parts?

Reply:
Ah got it. I also wonder if we want to move the substitution to `basic_auth_details` as I think technically this could also happen for base64-encoded tokens (that contain @ signs) too.

Reply:
Ah, yes good call! 👍